diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/BlockStructureBraces.kt

```diff
@@ -104,22 +106,28 @@ class BlockStructureBraces : Rule("block-structure") {
 
     private fun checkLoop(node: ASTNode, configuration: BlockStructureBracesConfiguration) {
         node.findChildByType(BODY)?.takeIf { body -> body.hasChildOfType(BLOCK) }?.let {
-            checkOpenBraceOnSameLine(node, BODY, configuration)
-            checkCloseBrace(it.findChildByType(BLOCK)!!, configuration)
-            if (node.elementType == DO_WHILE){
-                val allMiddleNode = listOf(node.findChildByType(BODY)!!.treeNext)
-                checkMidBrace(allMiddleNode, node, WHILE_KEYWORD)
+            if (!isNodeEmpty(it.findChildByType(BLOCK))) {
+                checkOpenBraceOnSameLine(node, BODY, configuration)
+                checkCloseBrace(it.findChildByType(BLOCK)!!, configuration)
+                if (node.elementType == DO_WHILE) {
+                    val allMiddleNode = listOf(node.findChildByType(BODY)!!.treeNext)
+                    checkMidBrace(allMiddleNode, node, WHILE_KEYWORD)
+                }
             }
         }
     }
 
     private fun checkWhen(node: ASTNode, configuration: BlockStructureBracesConfiguration) {
-        checkOpenBraceOnSameLine(node, LBRACE, configuration)
-        checkCloseBrace(node, configuration)
+        val children = node.getChildren(null).toList()
+        val childrenAfterLBrace = children.subList(children.indexOfFirst { it.elementType == LBRACE }, children.size)
```

Comment:
```suggestion
        val childrenAfterLBrace =  node.getChildren(null).toList().run { subList(indexOfFirst { it.elementType == LBRACE }, size) }
```
Not really mandatory here, but personally I like such style in kotlin: instead of creating variable and using it by name several times, you can wrap small transformations in methods like `run` and make code more compact.